src/kafkit/registry/sansio.py

```diff
@@ -511,22 +560,29 @@ class MockRegistryApi(RegistryApi):
     """
 
     DEFAULT_HEADERS = {
-        'content-type': "application/vnd.schemaregistry.v1+json"
+        "content-type": "application/vnd.schemaregistry.v1+json"
     }
 
-    def __init__(self, url='http://registry:8081',
-                 status_code=200, headers=None, body=b''):
+    def __init__(
+        self,
+        url: str = "http://registry:8081",
+        status_code: int = 200,
+        headers: Mapping[str, str] = None,
+        body: Any = b"",
+    ) -> None:
         super().__init__(url=url)
         self.response_code = status_code
         self.response_headers = headers if headers else self.DEFAULT_HEADERS
         self.response_body = body
 
-    async def _request(self, method, url, headers, body):
+    async def _request(
+        self, method: str, url: str, headers: Mapping[str, str], body: bytes
+    ) -> Any:
         self.method = method
         self.url = url
         self.headers = headers
         self.body = body
-        response_headers = self.response_headers.copy()
+        response_headers = dict(self.response_headers)  # make a copy
```

Comment:
Very minor nit: I tend to use `copy.deepcopy` in cases like this just to make the copy explicit.